app/optscale_api/helpers/auth_tokens_dependency.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+
+import logging
+
+from app.core.exceptions import UserAccessTokenError
+from app.optscale_api.auth_api import OptScaleAuth
+
+logger = logging.getLogger("helper")
+
+
+TOKEN_ERROR_MESSAGE = "Failed to obtain access token for user ID: {}"
+
+
+def get_auth_client() -> OptScaleAuth:
+    return OptScaleAuth()
```

Comment:
What does this function do? I don't think we need it

Reply:
Why not?

Reply:
Does it provide any benefit over just using `OptScaleAuth()`?

And for drawbacks:

* Increases the stack size needlessly making exceptions a bit harder to read and code harder to navigate
* If I want to change `get_auth_client()` to `get_auth_client(some_argument)`, I need to also change the function defintion
* Inevitable mixed usage of `get_auth_client` and `OptScaleAuth` will lead to harder to follow code

All minor but hey it's something :D What are you trying to do this here - I feel like I'm missing something here

Reply:
The `get_auth_client` is used as a dependency function.  I think that having it allows us to centralize logic. For example, in the future, the process of creating an instance of OptScaleAuth might change, and we need to pass some parameters, we can easily do something like 

```
def get_auth_client() -> OptScaleAuth:

      return OptScaleAuth(param=get_confi())
```
Moreover, FastAPI uses dependencies defined as functions, even for simple cases. This creates a declarative structure for its dep-injection pattern.

 TBH, I find this implementation easier to test and mock. It is much easier to override a dependency function than to replace a direct instantiation.
I don't find the code less readable or hard to navigate